File: python/examples/matched_name.py
# -*- coding: utf-8 -*-

'''
Example code to call Rosette API to get match score (similarity) of two names.
'''

import argparse
import pprint
import sys

# enable imports from rosette.api
sys.path += '../../'

from rosette.api import API, RniParameters

parser = argparse.ArgumentParser(description='Accept Rosette API key')
parser.add_argument('--key', required=True, help='Rosette API key')
parser.add_argument('--service_url', nargs='?', help='Optional user service URL')
args = parser.parse_args()

# Create an API instance
if args.service_url:
    api = API(service_url=args.service_url, user_key=args.key)
else:
    api = API(user_key=args.key)

params = RniParameters()
params["name1"] = { "text":"Michael Jackson", "language": "eng", "entityType":"PERSON" }
params["name2"] = { "text":"迈克尔·杰克逊", "entityType":"PERSON" }
#params["name1"] = { "text":"Elizabeth Doe", "language": "eng", "entityType":"PERSON" }
#params["name2"] = { "text":"Liz Doe", "language": "eng", "entityType":"PERSON" }
op = api.matched_name()
result = op.operate(params)

pprint.pprint(result)

File: nodejs/examples/matched-name.js
/*
 * Example code to call Rosette API to get match score (similarity) for two names.
 */

"use strict";

var Api = require("rosette-api").Api;
var ArgumentParser = require("argparse").ArgumentParser;
var NameMatchingParameters = require("rosette-api").NameMatchingParameters;

var parser = new ArgumentParser({
  addHelp: true,
  description: "Get the similarity score of two names"
});
parser.addArgument(["--key"], {help: "Rosette API key", required: true});
var args = parser.parseArgs();

var name1 = {"text": "${matched_name_data1}", "language": "eng", "entityType": "PERSON"};
var name2 = {"text": "${matched_name_data2}", "entityType": "PERSON"};
var matchParams = new NameMatchingParameters(name1, name2);

var api = new Api(args.key);
api.matchedName(matchParams, function(err, res) {
  if (err) {
    throw err;
  }
  else {
    console.log(res);
  }
});
